pkg/controller/appdefinition/namespace.go

```diff
@@ -82,3 +85,53 @@ func RequireNamespace(h router.Handler) router.Handler {
 		return h.Handle(req, resp)
 	})
 }
+
+func AddNamespace(req router.Request, resp router.Response) error {
```

Comment:
Does this function have to stay in appdefinition package?  Can it be moved out to controller/namespace.  I'd like to minimize the amount of code that we dump in appdefinition package.

Reply:
I can move it